libbpf-cargo/src/test.rs

```diff
@@ -2043,3 +2043,105 @@ pub struct Foo {
 
     assert_definition(&btf, struct_foo, expected_output);
 }
+
+#[test]
+fn test_btf_dump_definition_unnamed_union() {
+    let prog_text = r#"
+#include "vmlinux.h"
+#include <bpf/bpf_helpers.h>
+
+// re-typed 'struct bpf_sock_tuple tup' from vmlinux as of kernel 5.15
+// with a little bit added for additional complexity testing
+struct bpf_sock_tuple_5_15 {
+	union {
+		struct {
+			__be32 saddr;
+			__be32 daddr;
+			__be16 sport;
+			__be16 dport;
+		} ipv4;
+		struct {
+			__be32 saddr[4];
+			__be32 daddr[4];
+			__be16 sport;
+			__be16 dport;
+		} ipv6;
+	};
+
+    union {
+        int a;
+        char *b;
+    };
+};
+struct bpf_sock_tuple_5_15 tup;
+"#;
+
+    let expected_output = r#"
+#[derive(Debug, Default, Copy, Clone)]
+#[repr(C)]
+pub struct bpf_sock_tuple_5_15 {
+    pub __unnamed___anon_1: __anon_1,
```

Comment:
couldn't the field name be the same as type name? Rust doesn't allow this?

Reply:
Didn't think to try.

I think it would be best to not have the __unnamed_.  I shall fix later on today after work and have a review for tomorrow.

Reply:
Addressed in https://github.com/libbpf/libbpf-rs/pull/174.

Sorry this is later than I claimed.